engine/Shopware/Components/Translation.php

```diff
@@ -36,11 +36,48 @@ class Shopware_Components_Translation
     private $connection;
 
     /**
+     * @var int
+     */
+    private $language;
+
+    /**
+     * @var int
+     */
+    private $fallback;
+
+    /**
      * @param Connection|null $connection
+     * @param Container|null  $container
      */
-    public function __construct(Connection $connection = null)
+    public function __construct(Connection $connection = null, $container = null)
     {
-        $this->connection = $connection ?: Shopware()->Container()->get('dbal_connection');
+        $container = ($container) ?: Shopware()->Container();
+        $this->connection = $connection ?: $container->get('dbal_connection');
+
+        // Determine how to query the current language
+        if ($container->has('Auth')) {
```

Comment:
Please use the exact name of the services as defined in `services.xml` etc., because the DI container sometimes gets confused when getting the same service using a different name (capitalization) and creates multiple instances of the same service.

Reply:
That is the exact name. However, it does not refer to a service (defined in a service.xml) but rather to a Shopware default plugin.